src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Reflection/SignatureTypes.cs

```diff
@@ -48,10 +43,53 @@ public static IEnumerable<object[]> IsSignatureTypeTestData
                 yield return new object[] { typeof(List<>).MakeGenericType(sigType), true };
 
                 yield return new object[] { Type.MakeGenericSignatureType(typeof(List<>), typeof(int)), true };
+                yield return new object[] { Type.MakeGenericSignatureType(typeof(List<>), typeof(int)).GetGenericArguments()[0], false };
                 yield return new object[] { Type.MakeGenericSignatureType(typeof(List<>), sigType), true };
+                yield return new object[] { Type.MakeGenericSignatureType(typeof(List<>), sigType).GetGenericArguments()[0], true };
             }
         }
 
+        [Theory]
+        [MemberData(nameof(IsValueTypeTestData))]
+        public static void IsValueType(Type type, bool expected)
+        {
+            Assert.Equal(expected, type.IsValueType);
+        }
+
+        public static IEnumerable<object[]> IsValueTypeTestData
+        {
+            get
+            {
+                // Standard reflection used as baseline.
+                yield return new object[] { typeof(int), true };
+                yield return new object[] { typeof(object), false };
+                yield return new object[] { typeof(int).MakeArrayType(), false };
+                yield return new object[] { typeof(int).MakeByRefType(), false };
+                yield return new object[] { typeof(int).MakePointerType(), false };
+                yield return new object[] { typeof(List<>).GetGenericArguments()[0], false };
+                yield return new object[] { typeof(List<>).MakeGenericType(typeof(int)), false };
+                yield return new object[] { typeof(List<>).MakeGenericType(typeof(object)), false };
+                yield return new object[] { typeof(SignatureTypeTests).GetMethod(nameof(MyGenericValueTypeMethod), BindingFlags.NonPublic | BindingFlags.Static).GetGenericArguments()[0], true };
+                yield return new object[] { typeof(SignatureTypeTests).GetMethod(nameof(MyGenericRefTypeMethod), BindingFlags.NonPublic | BindingFlags.Static).GetGenericArguments()[0], false };
+                yield return new object[] { typeof(List<>).MakeGenericType(typeof(int)).GetGenericArguments()[0], true };
+                yield return new object[] { typeof(List<>).MakeGenericType(typeof(object)).GetGenericArguments()[0], false };
+
+                // SignatureTypes.
```

Comment:
```suggestion
```
These are test cases for regular runtime reflection. I do not think we need them for testing IsValueType property on signature types.

Reply:
Removed